addons/mail/models/mail_channel.py

```diff
@@ -605,7 +605,16 @@ def message_post(self, *, message_type='notification', **kwargs):
         if moderation_status == 'rejected':
             return self.env['mail.message']
 
-        self.filtered(lambda channel: channel.is_chat).mapped('channel_last_seen_partner_ids').sudo().write({'is_pinned': True})
+        if self.is_chat:
+            self.mapped('channel_last_seen_partner_ids').sudo().write({'is_pinned': True})
+            partner_ids = self.channel_partner_ids.ids
+            self.env['mail.channel.partner'].search([
+                ('partner_id', 'in', partner_ids),
+                ('channel_id', '=', self.id),
+            ]).write({
+                'last_activity_time': fields.Datetime.now()
+            })
```

Comment:
You could just write `last_activity_time` at the same time `is_pinned` instead of making a new search and write for it.

Reply:
`write()` method is overridden to update `last_meaningful_action_time` when `is_pinned` is modified